--- a/src/server/game/Server/Protocol/Handlers/DuelHandler.cpp	2011-07-20 04:59:50.000000000 +0000
+++ b/src/server/game/Server/Protocol/Handlers/DuelHandler.cpp	2011-07-20 05:33:58.000000000 +0000
@@ -51,6 +51,14 @@
 
     pl->SendDuelCountdown(3000);
     plTarget->SendDuelCountdown(3000);
+
+    // Reset CoolDown before DUEL
+    pl->RemoveAllSpellCooldown();
+    plTarget->RemoveAllSpellCooldown();
+    pl->SetHealth(pl->GetMaxHealth());
+    pl->SetPower(POWER_MANA, pl->GetMaxPower(POWER_MANA));
+    plTarget->SetHealth(plTarget->GetMaxHealth());
+    plTarget->SetPower(POWER_MANA,  plTarget->GetMaxPower(POWER_MANA));
 }
 
 void WorldSession::HandleDuelCancelledOpcode(WorldPacket& recvPacket)
