--- a/src/server/scripts/Custom/CMakeLists.txt	2011-07-20 04:59:50.000000000 +0000
+++ b/src/server/scripts/Custom/CMakeLists.txt	2011-07-20 06:05:52.000000000 +0000
@@ -10,6 +10,9 @@
 
 set(scripts_STAT_SRCS
   ${scripts_STAT_SRCS}
+  Custom/guildmaster.cpp
+  Custom/teleport.cpp
+  Custom/teleport_guild.cpp
 )
 
 message("  -> Prepared: Custom")

--- a/src/server/game/Scripting/ScriptLoader.cpp	2011-07-20 04:59:50.000000000 +0000
+++ b/src/server/game/Scripting/ScriptLoader.cpp	2011-07-20 06:44:29.000000000 +0000
@@ -17,6 +17,11 @@
 
 #include "ScriptLoader.h"
 
+//Custom scripts
+void AddSC_guild_teleporter();
+void AddSC_server_teleporter();
+void AddSC_guildmaster();
+
 //examples
 void AddSC_example_creature();
 void AddSC_example_escort();
@@ -677,6 +682,10 @@
     AddSC_npc_taxi();
     AddSC_achievement_scripts();
     AddSC_chat_log();
+    //Custom scripts
+    AddSC_guild_teleporter();
+    AddSC_server_teleporter();
+    AddSC_guildmaster();
 #endif
 }
