--- a/src/server/game/Spells/Auras/SpellAuraEffects.cpp	2011-08-09 16:47:00.000000000 +0000
+++ b/src/server/game/Spells/Auras/SpellAuraEffects.cpp	2011-08-11 12:53:39.000000000 +0000
@@ -592,7 +592,14 @@
                 break;
             // Earth Shield
             if (GetSpellInfo()->SpellFamilyName == SPELLFAMILY_SHAMAN && m_spellInfo->SpellFamilyFlags[1] & 0x400)
+            {
+                // return to unmodified by spellmods value
+                amount = m_spellInfo->Effects[m_effIndex].BasePoints;
+                // apply spell healing bonus
                 amount = caster->SpellHealingBonus(GetBase()->GetUnitOwner(), GetSpellInfo(), amount, SPELL_DIRECT_DAMAGE);
+                // apply spellmods
+                amount = caster->ApplyEffectModifiers(GetSpellInfo(), m_effIndex, float(amount));
+            }
             break;
         case SPELL_AURA_PERIODIC_DAMAGE:
             if (!caster)