--- a/src/server/game/Spells/Auras/SpellAuraEffects.cpp	2011-08-09 16:47:00.000000000 +0000
+++ b/src/server/game/Spells/Auras/SpellAuraEffects.cpp	2011-08-11 13:04:35.000000000 +0000
@@ -5192,21 +5199,37 @@
             if (!(mode & AURA_EFFECT_HANDLE_REAL))
                 break;
             // Sentry Totem
-            if (GetId() == 6495 && caster->GetTypeId() == TYPEID_PLAYER)
+            switch(GetId())
             {
-                if (apply)
+                // Sentry Totem (Serverside)
+                case 6494:
                 {
-                    uint64 guid = caster->m_SummonSlot[3];
-                    if (guid)
+                    if (Player * pCaster = caster->ToPlayer())
                     {
-                        Creature* totem = caster->GetMap()->GetCreature(guid);
-                        if (totem && totem->isTotem())
-                            caster->ToPlayer()->CastSpell(totem, 6277, true);
+                        pCaster->StopCastingBindSight();
+                        if (apply)
+                        {
+                            uint64 guid = pCaster->m_SummonSlot[4];
+                            if (guid)
+                            {
+                                Creature * totem = pCaster->GetMap()->GetCreature(guid);
+                                if (totem && totem->isTotem())
+                                    pCaster->CastSpell(totem, 6496, true);
+                            }
+                        }
+                        else
+                            pCaster->RemoveAurasDueToSpell(6495);
                     }
+                    break;
+                }
+                // Sentry Totem (hack for login case)
+                case 6495:
+                {
+                    if (caster->GetTypeId() == TYPEID_PLAYER && apply)
+                        if (!caster->m_SummonSlot[4])
+                            caster->RemoveAurasDueToSpell(6495);
+                    break;
                 }
-                else
-                    caster->ToPlayer()->StopCastingBindSight();
-                return;
             }
             break;
         }