--- a/src/server/game/Entities/Unit/Unit.cpp	2011-08-09 14:57:58.000000000 +0000
+++ b/src/server/game/Entities/Unit/Unit.cpp	2011-08-11 12:49:28.000000000 +0000
@@ -7883,6 +7883,13 @@
             }
             break;
         }
+        case SPELLFAMILY_WARRIOR:
+        {
+            // Recklessness - prevent double proc
+            if (dummySpell->Id == 1719)
+                return false;
+            break;
+        }
     }
     // processed charge only counting case
     if (!triggered_spell_id)